bot/exts/moderation/infraction/_utils.py

```diff
@@ -269,3 +272,44 @@ async def send_private_embed(user: MemberOrUser, embed: discord.Embed) -> bool:
             "The user either could not be retrieved or probably disabled their DMs."
         )
         return False
+
+
+class StaffBanConfirmationView(interactions.ViewWithUserAndRoleCheck):
+    """The confirmation view that is sent when a moderator attempts to ban a staff member."""
+
+    def __init__(
+        self,
+        *,
+        allowed_users: t.Sequence[int],
+        allowed_roles: t.Sequence[int],
+        timeout: t.Optional[float] = 180,
+        message: t.Optional[Message] = None
+    ) -> None:
+        super().__init__(
+            allowed_users=allowed_users,
+            allowed_roles=allowed_roles,
+            timeout=timeout,
+            message=message
+        )
+
+        self.value: t.Optional[bool] = None
+
+    async def send(self, channel: TextChannel, *, message_content: str) -> None:
+        """Send this view to the given channel with the given message content."""
+        self.message = await channel.send(message_content, view=self)
+
+    @discord.ui.button(label="Confirm", style=ButtonStyle.green)
+    async def confirm(self, interaction: Interaction, button: Button) -> None:
+        """Callback coroutine that is called when the "confirm" button is pressed."""
+        await interaction.response.defer()
```

Comment:
Is this defer needed?

Reply:
Yes. We're not directly responding to button's interaction, but sending a separate message. If we want the ban to be a response of the button interaction, we'd have to pass it into the `apply_ban` function, which would need to have updated tests and update all references. I didn't think it to be worth it.

Reply:
Yea, but when we do send the seperate message, we don't use the interaction to do so right?

Reply:
No, we don't. The message is just sent to the channel as a regular message. IIRC the interaction would "time out" even though we sent a message - since as far as Discord cares we didn't respond to their interaction

Reply:
AH, I see what is going on. Discord.py usually deals with the response to the callback automatically, but since you're calling self.timeout() manually, that s stopping the interaction, so Discord.y does not respond to Discord to say the interaction was handled.

I think what we want to do here is
```python
await interaction.message.edit(view=None)
self.stop()
```
At the end of both button callbacks You could even edit the embed at the same time for feedback to the person who interacted with the button. EG Update it to cancelled,/confirmed, or even delete the bot message and send an ephemeral message.